great_expectations/core/usage_statistics/anonymizers/anonymizer.py

```diff
@@ -1,765 +1,202 @@
-
-        return anonymized_info_dict
-
-    def anonymize_store_info(self, store_name: str, store_obj: object) -> dict:
-        """Anonymize Store objs from the 'great_expectations.data_context.store' module.
-
-        Args:
-            store_name (str): The name of the store.
-            store_obj (object): An instance of the Store base class or one of its children.
-
-        Returns:
-            An anonymized dictionary payload that obfuscates user-specific details.
-        """
-        anonymized_info_dict = {}
-        anonymized_info_dict["anonymized_name"] = self.anonymize(store_name)
-        store_backend_obj = store_obj.store_backend
-
-        self._anonymize_object_info(
-            object_=store_obj,
-            anonymized_info_dict=anonymized_info_dict,
+        from great_expectations.core.usage_statistics.anonymizers.datasource_anonymizer import (
+            DatasourceAnonymizer,
+        )
+        from great_expectations.core.usage_statistics.anonymizers.expectation_anonymizer import (
+            ExpectationSuiteAnonymizer,
+        )
+        from great_expectations.core.usage_statistics.anonymizers.profiler_anonymizer import (
+            ProfilerAnonymizer,
+        )
+        from great_expectations.core.usage_statistics.anonymizers.store_anonymizer import (
+            StoreAnonymizer,
+        )
+        from great_expectations.core.usage_statistics.anonymizers.store_backend_anonymizer import (
+            StoreBackendAnonymizer,
+        )
+        from great_expectations.core.usage_statistics.anonymizers.validation_operator_anonymizer import (
+            ValidationOperatorAnonymizer,
         )
 
-        anonymized_info_dict[
-            "anonymized_store_backend"
-        ] = self._anonymize_store_backend_info(store_backend_obj=store_backend_obj)
-
-        return anonymized_info_dict
-
-    def anonymize_profiler_info(self, name: str, config: dict) -> dict:
-        """Anonymize RuleBasedProfiler objs from the 'great_expectations.rule_based_profiler' module.
+        self._strategies: List[Type[BaseAnonymizer]] = [
+            CheckpointAnonymizer,
+            ProfilerAnonymizer,
+            DatasourceAnonymizer,
+            DataConnectorAnonymizer,
+            ActionAnonymizer,
+            DataDocsAnonymizer,
+            ExpectationSuiteAnonymizer,
+            ValidationOperatorAnonymizer,
+            BatchRequestAnonymizer,
+            BatchAnonymizer,
+            StoreAnonymizer,
+            StoreBackendAnonymizer,
+        ]
 
-        Args:
-            name (str): The name of the given profiler.
-            config (dict): The dictionary configuration of the given profiler.
+        # Instead of instantiating all child anonymizers, we perform JIT or on-demand instantiation.
+        # While this has some performance benefits, the primary reason is due to cyclic imports and recursive overflow.
+        self._cache: Dict[Type[BaseAnonymizer], BaseAnonymizer] = {}
```

Comment:
We have a few problems that this implementation solves:
1. Cyclic imports - Our "strategies" should definitely be class level attributes but there's no feasible way to make that work with the current state of the codebase. I've opted to place them in the constructor here to shield them from the global scope, allowing us to get by without issues.
2. Infinite recursion - if I were to instantiate all these child anonymizers right here and then use an instance of this anonymizer in the child classes, there would be infinite recursion.

Let me try to explain this as best as I can. Our primary `Anonymizer` class is sort of the web between all anonymizers in GE. While this class references specific implementations, those specific implementations also reference back here. This is deemed necessary because if a particular class requires behavior that is outside the responsibility of its class, it requests that behavior from this primary `Anonymizer` class. 

If my `DatasourceAnonymizer` has a constructor like this:
```python
def __init__(self, salt, aggregate_anonymizer):
    ...
```
it becomes evident that part of the definition of the class is its relationship with the primary `Anonymizer`. This bidrectional link causes recursion issues because both classes are defined in terms of each other. Instead, I've opted to only instantiate what we need at a given time and cache those results. This could probably be explained better live but this solution here seems to have mitigated much of the pain of this refactor.